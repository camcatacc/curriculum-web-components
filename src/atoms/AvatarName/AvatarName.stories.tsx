import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import AvatarName from "atoms/AvatarName/AvatarName";

const Template: ComponentStory<typeof AvatarName> = (args) => <AvatarName {...args} />;

export const Standard = Template.bind({});
Standard.args = {
	name: "Name",
	surname: "Surname1 Surname2"
};

export default {
	title: "Atoms/AvatarName",
	component: AvatarName
} as ComponentMeta<typeof AvatarName>;
