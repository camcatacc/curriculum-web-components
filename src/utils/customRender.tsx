import { ThemeProvider } from "@material-ui/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { ReactElement } from "react";
import { getMaterialTheme } from "styles/materialUi";

export const customRender = (ui: ReactElement, renderOptions?: RenderOptions & { darkMode?: boolean }): RenderResult => {
	const Wrapper: React.FC = ({ children }) => {
		return <ThemeProvider theme={getMaterialTheme(renderOptions?.darkMode ? "dark" : "light")}>{children}</ThemeProvider>;
	};

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};
