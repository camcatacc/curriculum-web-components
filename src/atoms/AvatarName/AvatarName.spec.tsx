// Modules
import React from "react";
import { customRender } from "utils/customRender";

// Accessibility
import { axe, toHaveNoViolations } from "jest-axe";

// Elements
import AvatarName from "atoms/AvatarName/AvatarName";

// Tests
expect.extend(toHaveNoViolations);
describe("AvatarName", () => {
	it("Renders as expected and has no accessibility violations", async () => {
		const { container } = customRender(<AvatarName name="Marco" surname="Polo" />);
		expect(container).toMatchSnapshot();
		expect(await axe(container)).toHaveNoViolations();
	});
});
