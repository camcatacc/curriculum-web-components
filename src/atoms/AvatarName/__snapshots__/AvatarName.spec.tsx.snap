// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarName Renders as expected and has no accessibility violations 1`] = `
<div>
  <h3
    class="MuiTypography-root sc-bdvvtL gjaeEB MuiTypography-h3 MuiTypography-alignCenter"
  >
    Marco
     
    Polo
  </h3>
</div>
`;
